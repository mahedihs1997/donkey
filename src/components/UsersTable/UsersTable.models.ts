export interface User {
  username: string;
  name: string;
}

export interface UsersTableProps {
  data: User[];
}
