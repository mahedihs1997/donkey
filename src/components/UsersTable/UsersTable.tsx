import React from 'react';
import { UsersTableProps } from './UsersTable.models';

import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import { useStyles } from './UsersTable.styles';

const UsersTable: React.FunctionComponent<UsersTableProps> = (props) => {
  const { data } = props;

  const classes = useStyles();

  return (
    <TableContainer component={Paper}>
      <Table className={classes.table} aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>Name</TableCell>
            <TableCell>Username</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {data.map((row, index) => (
            <TableRow key={`${row.name}-${index}`}>
              <TableCell component="th" scope="row">
                {row.name}
              </TableCell>
              <TableCell>{row.username}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
};

export default UsersTable;
