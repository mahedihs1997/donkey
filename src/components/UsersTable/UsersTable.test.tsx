import React from 'react';
import TestRenderer from 'react-test-renderer';
import UsersTable from '.';

const data = [
  { username: 'Ugendo', name: 'Uga' },
  { username: 'Bugendo', name: 'Buga' },
];

describe('<UsersTable />', () => {
  it('should render without throwing an error and match snapshot', () => {
    const testRenderer = TestRenderer.create(<UsersTable data={data} />);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
