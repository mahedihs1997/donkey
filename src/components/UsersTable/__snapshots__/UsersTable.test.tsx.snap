// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersTable /> should render without throwing an error and match snapshot 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <table
    aria-label="simple table"
    className="MuiTable-root makeStyles-table-1"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Username
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          role="cell"
          scope="row"
        >
          Uga
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          Ugendo
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          role="cell"
          scope="row"
        >
          Buga
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          Bugendo
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
